src/Model/Action.php

```diff
@@ -0,0 +1,228 @@
+<?php
+
+declare(strict_types=1);
+
+namespace atk4\data\Model;
+
+use atk4\core\DIContainerTrait;
+use atk4\core\Exception;
+use atk4\core\InitializerTrait;
+use atk4\core\TrackableTrait;
+use atk4\data\Model;
+
+/**
+ * Implements generic user action. Assigned to a model it can be invoked by a user. Model\Action class contains a
+ * meta information about the action (arguments, permissions, scope, etc) that will help UI/API or add-ons to display
+ * action trigger (button) correctly in an automated way.
+ *
+ * Action must NOT rely on any specific UI implementation.
+ *
+ * @property Model $owner
+ */
+class Action
+{
+    use DIContainerTrait;
+    use TrackableTrait;
+    use InitializerTrait {
+        init as init_;
+    }
+
+    /** Defining scope of the action */
+    const SCOPE_NONE = 'none'; // e.g. add
```

Comment:
wdyt about _NEW instead? Is there any other use case? (my reasoning: if not new, then it is single :) )

Reply:
We may not grasp the range of possibilities so I would keep it with original design 'NONE'.
'NEW' is just a single case of 'NONE' and will limits us.

Reply:
The question is - is there any other case then NEW?

Reply:
?

Reply:
Questuon is "could there be?". And we don't know.
So better keep it open.
Anyway this word refers to the number of records. We should not mix it with particular names of actions.
Once this refactoring series is complete I will seek to extend the functionality of Action 'scope' to accept Model\Scope object as well so action is applied to records in the scope only.

Reply:
Ok - in sense of this PR not critical...